### ISSUE TITLE
Bug: `array-element-newline` fails when `multiline` = `true`

### ISSUE THREAD

**Issue by bsara** [eslint/eslint#8981]
**Tell us about your environment**

* **ESLint Version:** 4.3.0
* **Node Version:** *
* **npm Version:** *

**What parser (default, Babel-ESLint, etc.) are you using?**
I had this happen locally, but I've been able to duplicate it using http://eslint.org/demo

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  rules: {
    "array-element-newline": [ "error", { "multiline": true } ]
  }
}
```

</details>

<br/>

**What did you do? Please include the actual source code causing the issue.**

```js
var blah = [
  0,
  1
];
```

**What did you expect to happen?**
`array-element-newline` rule should succeed

**What actually happened? Please include the actual, raw output from ESLint.**

The evaluated to a failure.

Output:
```
4:5 - There should be no linebreak here. (array-element-newline)
```

<br/>

---

<br/>

This can be easily reproduced if you go to http://eslint.org/demo, uncheck all rules, and insert the following text into the input box:

```js
/*eslint array-element-newline: ["error", { "multiline": true }]*/

var blah = [
  0,
  1
];
```

**Comment by platinumazure** [comment]
I think this is working as designed. The [multiline option](http://eslint.org/docs/rules/array-element-newline#options) is designed to warn on the rule if the *element* is multiline, not if the array is multiline.

That said, I'm not sure how one is supposed to configure that rule to require linebreaks if the entire array is multiline...

**Comment by bsara** [comment]
Wow, I didn't notice that at all. Sorry to bug you guys with this.

Perhaps `multiline` should be renamed to `multiline-element` to avoid confusion. It actually took reading the doc a few times before I really caught the part that says *"if there are line breaks inside elements"*.

Also, a multi-line option that enforces when any element is on a newline would be useful.

**Comment by not-an-aardvark** [comment]
> Also, a multi-line option that enforces when any element is on a newline would be useful.

We considered this when creating the rule (https://github.com/eslint/eslint/issues/8375). The problem is that this ends up creating a circular logic where the rule wouldn't actually be enforcing anything. The option would basically be saying "if there are newlines between elements, then the array is multiline, so enforce newlines between elements", which would pass validation regardless of whether there are actually newlines.

**Comment by ilyavolodin** [comment]
Is there any follow-up necessary for this issue? Maybe we can improve documentation somehow, to make `multiline` constrains a bit more noticeable?

**Comment by bsara** [comment]
@ilyavolodin, yes. IMO, the documentation really needs better wording to point out when the option is really for.

**Comment by bsara** [comment]
Also, it would be wonderful if by some means, a way could be found to support instances where the rule is only enforced when a at least one array element is on a new line. (like the functionality of [stylelint](https://stylelint.io/user-guide/rules/selector-list-comma-newline-after/) in many of it's rules. I know that stylelint may be doing things drastically different from ESLint, but I mention it only as an example of the functionality that I would like).

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended, but feel free to create a PR if you think the documentation can be improved.

**Comment by richardpringle** [comment]
So... I might just be going crazy here but I swear you used to be able to enforce a rule where elements cannot overlap.

The most common case (maybe the only case) that I'm looking to solve is as follows:
```node
const arr = [{
    hello: 'world'
  }, {
    foo: 'bar'
}]
```

What I would like to see is the ability to enforce that all array elements either share a single line or that no elements share lines. An oversimplification of this is that every line in a multiline array should end with a comma (except the last, depending on comma dangle).
